src/main/java/com/gruzini/tennistico/listeners/NotificationListener.java

```diff
@@ -0,0 +1,23 @@
+package com.gruzini.tennistico.listeners;
+
+import com.gruzini.tennistico.events.NotificationEvent;
+import com.gruzini.tennistico.services.NotificationService;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+@Component
+public class NotificationListener {
+
+    private final NotificationService notificationService;
+
+    public NotificationListener(NotificationService notificationService) {
+        this.notificationService = notificationService;
+    }
+
+    @Async
+    @EventListener
+    public synchronized void handleNotificationPublish(final NotificationEvent event) {
```

Comment:
@serwacy @bartekmat  Myślicie, że ta metoda powinna być synchronized?
Bo tak mnie teraz naszło, że może niekoniecznie

Reply:
ważne chyba, może się zdarzyć sytuacja, że ktoś zjoinuje HOSTED grę i w czasie zanim Publisher zadziała, ktoś wybierze tę samą grę i wtedy może się chyba zrobić kupa jakby nie było synchronized

Reply:
hmm, a czy osobny wątek nie załatwi już tej sprawy? Grzesiu - po dołączeniu do hosted match zmienia on status i przestaje być widoczny. Jak ktoś inny ma na od 5 min otwartą listę gier i kliknie join to powinno go wywalić wogóle a nie robić jakieś synchronized. Musimy to zabezpieczyć żeby dostał error page - 'ktoś inny już dołączył, gra niedostępna'

Reply:
mówię tutaj o bardzo wyjątkowym przypadku, kiedy dwie osoby klikną ten sam mecz praktycznie w tym samym momencie i wtedy nie pozwoli wątkowi wejść do metody jeśli przed chwilą wszedł tam inny i jeszcze ją procesuje

Reply:
Czyli trzeba by wpuszczać jeden wątek i na starcie, już po wejściu do metody, walidować czy można taką zmianę wykonać. Niekoniecznie to w wysyłaniu notyfikacji powinna być - raczej w zmienianiu statusu gry. Jak nie można zmienić statusu, to exception i error page zwracany przez handler. Dobrze kminię?